File: email-backup.java/src/main/java/demo/Configuration.java
package demo;

import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.GnuParser;
import org.apache.commons.cli.HelpFormatter;
import org.apache.commons.cli.Options;
import org.apache.commons.cli.ParseException;

final public class Configuration {

    public static Flags parseFlags(String... args) {
        Options options = OptionsBuilder.build();

        try {
            CommandLine line = new GnuParser().parse(options, args);

            String email = (String) line.getParsedOptionValue("email");
            String password = (String) line.getParsedOptionValue("password");
            String host = (String) line.getParsedOptionValue("host");
            int port = ((Number) line.getParsedOptionValue("port")).intValue();
            String[] excludedExtensions = line.getOptionValues("excluded_extensions");
            String[] includedFiles = line.getOptionValues("included_files");
            String[] emailsTo = line.getOptionValues("emails_to");

            return new Flags(email, password, host, port, excludedExtensions, includedFiles, emailsTo);
        } catch (ParseException e) {
            System.out.println(e.getMessage());

            new HelpFormatter().printHelp("java -jar email-backup.jar", options);
            return null;
        }
    }
}

File: email-backup.kotlin/src/main/kotlin/demo/Configuration.kt
package demo

import org.apache.commons.cli.GnuParser
import org.apache.commons.cli.HelpFormatter
import org.apache.commons.cli.ParseException

public object Configuration {

    public fun parseFlags(vararg args: String): Flags? {
        val options = OptionsBuilder.build()

        try {
            val line = GnuParser().parse(options, args)

            val email = line.getParsedOptionValue("email") as String
            val password = line.getParsedOptionValue("password") as String
            val host = line.getParsedOptionValue("host") as String
            val port = (line.getParsedOptionValue("port") as Number).toInt()
            val excludedExtensions = line.getOptionValues("excluded_extensions")
            val includedFiles = line.getOptionValues("included_files")
            val emailsTo = line.getOptionValues("emails_to")

            return Flags(email, password, host, port, excludedExtensions, includedFiles, emailsTo)
        } catch (e: ParseException) {
            println(e.getMessage())

            HelpFormatter().printHelp("java -jar email-backup.jar", options)
            return null
        }
    }
}
